src/js/common/schemaform/ArrayField.jsx

```diff
@@ -164,25 +164,35 @@ export default class ArrayField extends React.Component {
 
     const title = uiSchema['ui:title'] || schema.title;
     const hideTitle = !!_.get(['ui:options', 'hideTitle'], uiSchema);
-    const hasTextDescription = typeof uiSchema['ui:description'] === 'string';
-    const DescriptionField = !hasTextDescription && typeof uiSchema['ui:description'] === 'function'
+    const description = uiSchema['ui:description'];
```

Comment:
I pushed up the changes in this file to fix some array field header spacing issues. There's a comment below about handling the array field title that would need these changes.